cloud-sql/mysql/ConnectUnix.cs

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+// [START cloud_sql_mysql_dotnet_ado_connect_unix]
+using MySql.Data.MySqlClient;
+using System;
+
+namespace CloudSql
+{
+    public class MySqlUnix
+    {
+        public static MySqlConnectionStringBuilder NewMysqlUnixSocketConnectionString()
+        {
+            // Equivalent connection string:
+            // "Server=<dbSocketDir>/<INSTANCE_CONNECTION_NAME>;Uid=<DB_USER>;Pwd=<DB_PASS>;Database=<DB_NAME>;Protocol=unix"
+            String dbSocketDir = Environment.GetEnvironmentVariable("INSTANCE_UNIX_SOCKET") ?? "/cloudsql";
```

Comment:
Let's move these down below. Also, "INSTANCE_UNIX_SOCKET" should be the full path and so we should remove `/cloudsql`.

Reply:
Done.